buildSrc/src/main/minimumRuntime/org/elasticsearch/gradle/Version.java

```diff
@@ -124,4 +124,7 @@ public int compareTo(Version other) {
         return Integer.compare(getId(), other.getId());
     }
 
+    public boolean isNextMajor() {
+        return getRevision() == 0 && getMinor() == 0;
```

Comment:
This doesn't seem right. A version string without context cannot possibly determine if it is the next major version. This should really live in `BwcVersions`. For example `Version.fromString("7.0.0").isNextMajor()` would return `true` which is obviously incorrect.

Reply:
The method shouldn't exist probably, and we should just do the check directly. 
BWC versions doesn't know about newer versions, e.x on `6.8` there's no way to know how far we'we got. That being said we have used `x.0.0` to identify master, as we will never build ex `7.0.0` as soon as it's released we version bump and the branch becomes  `7.0.1`.  This would break down only when we have a major  staged (as we had before `7.0` was released but the `7.x` branch cut). This would I think be acceptable for this use-case (without the `isNextMajor` method of course). WDYT @mark-vieira ?

Reply:
I'd be ok with renaming this something more pedantically accurate like `isNewMajor()`.